typescript-carousel/main.ts

```diff
@@ -0,0 +1,107 @@
+const $currentImage = document.querySelector(
+  '.current-image'
+) as HTMLImageElement;
+const $carousel = document.querySelector('.carousel') as HTMLDivElement;
+const $selectCircles = document.querySelector(
+  '.select-circles'
+) as HTMLDivElement;
+const $img1 = document.querySelector('.img1') as HTMLElement;
+const $img2 = document.querySelector('.img2') as HTMLElement;
+const $img3 = document.querySelector('.img3') as HTMLElement;
+const $img4 = document.querySelector('.img4') as HTMLElement;
+const $img5 = document.querySelector('.img5') as HTMLElement;
+
+if (!$currentImage) throw new Error('$currentImage query failed');
+if (!$carousel) throw new Error('$carousel query failed');
+if (!$selectCircles) throw new Error('$selectCircles query failed');
+if (!$img1) throw new Error('$img1 query failed');
+if (!$img2) throw new Error('$img2 query failed');
+if (!$img3) throw new Error('$img3 query failed');
+if (!$img4) throw new Error('$img4 query failed');
+if (!$img5) throw new Error('$img5 query failed');
+
+const images = [
+  'images/001.png',
+  'images/004.png',
+  'images/007.png',
+  'images/025.png',
+  'images/039.png',
+];
+const imgDots = [$img1, $img2, $img3, $img4, $img5];
+let i = 0;
+let setIntervalId = setInterval(cycleImages, 3000);
+
+function cycleImages(): void {
+  i++;
+  handleDots();
+  if (i === 5) {
```

Comment:
Rather than hard coding this, you should utilize the length of something. What if I want to add more images?